api/handlers.go

```diff
@@ -1109,48 +1110,77 @@ order by type_id, film_date desc
 		cIDs = append(cIDs, firstRows[consts.CT_HOLIDAY][0].id)
 		cs = append(cs, firstRows[consts.CT_HOLIDAY][0])
 	}
+
+	//last Collections: LESSONS_SERIES
+	clsIDs := make([]int64, 0)
+	cls := make([]firstRowsType, 0)
 	if _, ok := firstRows[consts.CT_LESSONS_SERIES]; ok {
 		// The first one is always on HomePage
 		for _, r := range firstRows[consts.CT_LESSONS_SERIES][0:2] {
 			cIDs = append(cIDs, r.id)
-			cs = append(cs, r)
+			clsIDs = append(clsIDs, r.id)
 		}
 	}
 
-	var lastNumber int
-	var lastDate time.Time
-
 	if _, ok := firstRows[consts.CT_DAILY_LESSON]; ok {
 		// The first one is always on HomePage
-		lastNumber = 1
-		lastDate = firstRows[consts.CT_DAILY_LESSON][0].film_date
 		for _, r := range firstRows[consts.CT_DAILY_LESSON][1:3] {
 			cIDs = append(cIDs, r.id)
-			cs = append(cs, r)
+			clsIDs = append(clsIDs, r.id)
+			cls = append(cls, r)
 		}
 	}
 
 	ci18nsMap, err := loadCI18ns(db, r.Language, cIDs)
 	if err != nil {
 		return nil, NewInternalError(err)
 	}
+
+	ct := make([]*ContentUnit, 0)
+
 	for _, x := range cs {
 		u := &ContentUnit{
 			mdbID:       x.id,
 			ID:          x.uid,
 			ContentType: x.content_type,
 			FilmDate:    &utils.Date{Time: x.film_date},
 		}
-		if x.content_type == consts.CT_DAILY_LESSON {
-			if x.film_date == lastDate {
-				lastNumber++
-			} else {
-				lastNumber = 1
-				lastDate = x.film_date
+		ct = append(ct, u)
+	}
+
+	// data query
+	cols, err := mdbmodels.Collections(db, qm.WhereIn("id IN ?", utils.ConvertArgsInt64(clsIDs)...)).All()
+	if err != nil {
+		return nil, NewInternalError(err)
+	}
+	for _, x := range cols {
+		u := &ContentUnit{
```

Comment:
Shouldn't this be `Collection` instead of `ContentUnit` ?
Or the db query above is for units ? I'm confused

Reply:
This that was heppend befor, as I undestand. 
Do you want that I separate units and collections?

Reply:
I don't think you understand my question.
I'm asking why are you instantiating `ContentUnit` structs? 
`cols` holds `models.Collection` so you should use `Collection` struct (dto)